text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+
+### Importing User Pools
+
+Any user pool that has been created outside of this stack, can be imported into the CDK app. Importing a user pool
+allows for it to be used in other parts of the CDK app that reference an `IUserPool`. However, imported user pools have
+limited configurability. As a rule of thumb, none of the properties that is are part of the
+[`AWS::Cognito::UserPool`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html)
+CloudFormation resource can be configured.
+
+User pools can be imported either by specifying the user pool id, via the `UserPool.fromUserPoolId()` API, or by
+specifying the user pool ARN, via the `UserPool.fromUserPoolArn()` API.
+
+```ts
+const stack = new Stack(app, 'my-stack');
+
+const awesomePool = UserPool.fromUserPoolId(stack, 'awesome-user-pool', 'us-east-1_oiuR12Abd');
+
+const otherAwesomePool = UserPool.fromUserPoolArn(stack, 'other-awesome-user-pool',
+  'arn:aws:cognito-idp:eu-west-1:123456789012:userpool/us-east-1_mtRyYQ14D');
+```
+
+### Users
+
+Users are added to pools either as part of sign up or via the Cognito APIs directly. In addition to these, it is
+possible to add users to the user pool via the CDK.
+
+Use the `createUser()` API on the user pool to add users to the pool, as so -
+
+```ts
+const userpool = new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+});
+
+userpool.createUser(this, 'myuserpool-supportuser', {
+  userName: 'support-user',
+  deliveryMediums: [ ContactMethods.EMAIL, ContactMethods.PHONE ],
+  attributes: [
+    { name: StandardAttrs.EMAIL, value: 'support@awesomeapp.com' },
+    { name: 'callingcode' value: '12' }
+  ],
+  forceAliasCreation: true,
+  resendInvitation: true,
+});
+```
+
+The property `userName` is the name of the user and is mandatory.
+
+The property `deliveryMedium` is how the user will be contacted to sign up, upon first creation. Both email and sms are
+available here.
+
+The property `attributes` specifies the list of standard and custom [attributes](#attributes) as part of the user.
+Values of the `StandardAttrs` enum can be used against `name` to reference standard attributes. If one or both contact
+methods are set with `deliveryMedium`, this attribute is required to be set as part of `attributes`.
+
+`forceAliasCreation` can be used only when the user pool is configured with either email or phone as a verified contact
+method. When this is set, any existing user with the same email address or phone number will have its alias migrated to
+this user.
+
+When the `resendInvitation` property is set to true, Cognito will resend the invitation message for a user that already
+exists but has not signed up.
+
+Users can be created for imported user pools by using the `UserPoolUser` construct, as so -
+
+```ts
+const userpool = UserPool.fromUserPoolAttributes(this, 'myuserpool', { ... });
+
+new UserPoolUser(this, userpool, {
+  userPoolId: userpool.userPoolId,
```

Comment:
Don't use ID, reference by object:

```ts
new UserPoolUser(this, userpool, {
  userPool: userpool,
});
```

Reply:
Whoops, that was when my brain took a break.